omnibus/package-scripts/agent/preinst

```diff
@@ -70,18 +70,33 @@ if [ "$DISTRIBUTION" != "Darwin" ]; then
                 -c "Datadog Agent" dd-agent && \
                 { usermod -L dd-agent || echo "[ WARNING ]\tCannot lock the 'dd-agent' user account"; }
 
-        # Uninstall every datadog integration
+
         # Starting with 6.10, integrations are also uninstalled on package removal
-        # See https://github.com/DataDog/datadog-agent/pull/3066 for more details
-        PIP2_PATH=$INSTALL_DIR/embedded/bin/pip2
-        PIP3_PATH=$INSTALL_DIR/embedded/bin/pip3
-        if [ -x $PIP2_PATH ]; then
-            echo "Uninstalling Python 2 integrations..."
-            $PIP2_PATH freeze | grep ^datadog- | grep -v datadog-checks-base | xargs $PIP2_PATH uninstall -y -q --no-cache-dir || true
-        fi
-        if [ -x $PIP3_PATH ]; then
-            echo "Uninstalling Python 3 integrations..."
-            $PIP3_PATH freeze | grep ^datadog- | grep -v datadog-checks-base | xargs $PIP3_PATH uninstall -y -q --no-cache-dir || true
+
+        # Since 6.18.0, a file containing all integrations files which have been installed by
+        # the package is available. We use it to remove only the datadog-related check files which
+        # have *NOT* been installed by the package (eg: installed using the `integration` command).
+
+        if [ -f "$INSTALL_DIR/embedded/.installed_by_pkg.txt" ]; then
+            # List all files in the embedded dir of the datadog-agent install dir
+            find $INSTALL_DIR/embedded/lib/python*/site-packages/datadog_* > $INSTALL_DIR/embedded/.all-integrations.txt
+
+            # List all files in the embedded dir of the datadog-agent install dir
+            # which were not installed by the package and rm them.
+            grep -Fxv -f $INSTALL_DIR/embedded/.installed_by_pkg.txt $INSTALL_DIR/embedded/.all-integrations.txt | grep -v '^#' | xargs rm -rf
```

Comment:
Call me paranoid but this has been taken from my PR, and I've hated writing this `rm -rf`.
Maybe we should call a delete method not using `-f` and validating that we're suppressing files owned by the `dd-agent` user?

Reply:
`rm -r` should work, but I'm not sure this makes a difference since this runs as root... it's still dangerous.

A safer alternative would be to `xargs -I '{}' rm -r $INSTALL_DIR/{}` to make sure we are constrained to that path, and remove the $INSTALL_DIR prefix from the files listed in the .txt's. Wdyt?

Reply:
Not sure that's safer because for whatever reasons we have an empty $INSTALL_DIR here, it would result with `rm -r /{}`...
What do you think of using something like (I've quickly drafted it, may contains error):
```
function delete_dd_agent_file() {
  if [ "$1" = "" ]; then
    return
  fi
  for f in $@; do
    owner=$(stat -c '%U' ${f})
    if [ "${owner}" = "dd-agent" ]; then
      rm -r "${f}"
    fi
  done
}
```
It would not work with the current grep one-liner because of the `\n` and the end of the filenames though. Not sure that's the way to go but I don't have anything better in mind for the moment. I'll let you have the final call.

Reply:
Actually, I didn't check if other RPM packages faced the same issue (they must have) and how did they resolve it.

Reply:
I think it's quite safe to assume that `$INSTALL_DIR` is not empty (if we want to be extra-cautious though, we can check it's not empty). Constraining to `$INSTALL_DIR` sounds like a good idea.

The main thing that worries me in this logic is that we search filenames (which can potentially be "malicious") and delete them in multiple steps, which is much scarier than doing it in one `find` command.
For instance, I would triple-check what happens if some of the file names in `$INSTALL_DIR/embedded/.all-integrations.txt` have spaces or newlines. For example, if there's a file named `some/path / with/spaces`, we obviously wouldn't want that to be passed to `rm -rf` as 3 arguments, i.e. `rm -rf some/path / with/spaces`. I haven't fully tested but I think it could happen with the current code.

"Malicious" filename I'd test:
```
/opt/datadog-agent/embedded/lib/python2.7/site-packages/datadog_tricky / tricky\n/\nfilename
```
(`\n` being actual newlines)